src/pynwb/file.py

```diff
@@ -618,15 +618,21 @@ def add_stimulus_template(self, timeseries):
              'help': 'description for table. Only used when passing in pandas.DataFrame', 'default': None})
     def add_scratch(self, **kwargs):
         '''Add data to the scratch space'''
-        data = getargs('data', kwargs)
+        data, name, notes, table_description = getargs('data', 'name', 'notes', 'table_description', kwargs)
         if isinstance(data, (np.ndarray, pd.DataFrame, list, tuple)):
-            name, notes, table_description = getargs('name', 'notes', 'table_description', kwargs)
             if name is None:
                 raise ValueError('please provide a name for scratch data')
             if isinstance(data, pd.DataFrame):
                 data = DynamicTable.from_dataframe(df=data, name=name)
```

Comment:
`table_description` should be passed in here. See [DynamicTable.from_dataframe](https://pynwb.readthedocs.io/en/latest/pynwb.core.html?highlight=DynamicTable#pynwb.core.DynamicTable.from_dataframe) for details.